Calculating price predictions on data using XGBoostRegressor

In [1]:
import pandas as pd

import numpy as np

In [2]:
df_for_predictions = pd.read_csv('/Users/baytech11/Desktop/Agnes/Springboard/Capstone Project/prepared data/final_data_unsclaed')

In [3]:
df_for_predictions.head(3)

Lot Area (sqf)  Property Price ($)  Number of Bedrooms  longitude  \
0        3001.284              995000                 2.0 -122.46822   
1        3125.000              898000                 4.0 -122.45815   
2        2495.988              995000                 4.0 -122.38945   

   latitude  Number of Bathrooms  Living Area (sqf)  Price per sqf ($)  \
0  37.75599                  1.0             1038.0              959.0   
1  37.71772                  2.0             1845.0              487.0   
2  37.73663                  4.0             2430.0              409.0   

   Median RE Price in the Area  Median Price Growth (%) in the Area  ...  \
0                    1422500.0                                 -6.7  ...   
1                    1422500.0                                 -6.7  ...   
2                    1422500.0                                 -6.7  ...   

   Construction Type_BUILDER_SPEC  Construction Type_False  \
0                               0                        1   
1                               0                        1   
2                               0                        1   

   Construction Type_NEW_CONSTRUCTION_TYPE_OTHER  Listing Type_True  \
0                                              0                  0   
1                                              0                  0   
2                                              0                  0   

   County_Contra Costa  County_Marin  County_San Francisco  County_San Mateo  \
0                    0             0                     1                 0   
1                    0             0                     1                 0   
2                    0             0                     1                 0   

   County_Santa Clara  County_Solano  
0                   0              0  
1                   0              0  
2                   0              0  

[3 rows x 56 columns]

In [4]:
import xgboost as xg

In [5]:
from xgboost import XGBRegressor
from sklearn.model_selection import train_test_split

In [6]:
X= df_for_predictions.drop("Price per sqf ($)", axis=1).values

y = df_for_predictions['Price per sqf ($)'].values

In [7]:
X_train, X_test, y_train, y_test = train_test_split(X,y,test_size = 0.2, random_state = 50)

In [8]:
xgb = XGBRegressor()


In [9]:
xgb.fit(X_train, y_train)

XGBRegressor(base_score=0.5, booster='gbtree', callbacks=None,
             colsample_bylevel=1, colsample_bynode=1, colsample_bytree=1,
             early_stopping_rounds=None, enable_categorical=False,
             eval_metric=None, feature_types=None, gamma=0, gpu_id=-1,
             grow_policy='depthwise', importance_type=None,
             interaction_constraints='', learning_rate=0.300000012, max_bin=256,
             max_cat_threshold=64, max_cat_to_onehot=4, max_delta_step=0,
             max_depth=6, max_leaves=0, min_child_weight=1, missing=nan,
             monotone_constraints='()', n_estimators=100, n_jobs=0,
             num_parallel_tree=1, predictor='auto', random_state=0, ...)

In [10]:
xgb.score(X_test, y_test)

0.8936063848553768

In [11]:
xgb_predict = xgb.predict(X_test[0:10])

In [12]:
xgb_predict

array([ 548.7229 ,  923.96606,  517.7968 ,  757.938  ,  547.6846 ,
       1419.243  ,  943.37067,  418.53745, 1049.437  ,  749.69696],
      dtype=float32)

In [13]:
predictions = pd.DataFrame(X_test[0:10], columns = [df_for_predictions.drop("Price per sqf ($)", axis=1)])

In [16]:
predictions.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 10 entries, 0 to 9
Data columns (total 55 columns):
 #   Column                                            Non-Null Count  Dtype  
---  ------                                            --------------  -----  
 0   (Lot Area (sqf),)                                 10 non-null     float64
 1   (Property Price ($),)                             10 non-null     float64
 2   (Number of Bedrooms,)                             10 non-null     float64
 3   (longitude,)                                      10 non-null     float64
 4   (latitude,)                                       10 non-null     float64
 5   (Number of Bathrooms,)                            10 non-null     float64
 6   (Living Area (sqf),)                              10 non-null     float64
 7   (Median RE Price in the Area,)                    10 non-null     float64
 8   (Median Price Growth (%) in the Area,)            10 non-null     float64
 9   (Median Home Price,)    

In [15]:
pd.set_option('display.max_columns', None)
predictions.head(10)

Lot Area (sqf) Property Price ($) Number of Bedrooms   longitude   latitude  \
0          2.200          1040000.0                3.0 -121.770640  37.878666   
1       8507.268          2850000.0                4.0 -122.328766  37.543877   
2       3049.200           825000.0                3.0 -122.081040  37.996620   
3       2996.928          1850000.0                4.0 -122.491844  37.763607   
4       4791.600           999000.0                4.0 -122.274445  37.860027   
5       5928.516          2398000.0                3.0 -122.337780  37.569584   
6       2613.600           949000.0                2.0 -122.251686  37.737747   
7        566.280          1025000.0                5.0 -122.107560  37.998333   
8       5662.800          2800000.0                4.0 -122.233505  37.750153   
9          0.000           759850.0                2.0 -122.021095  37.368046   

  Number of Bathrooms Living Area (sqf) Median RE Price in the Area  \
0                 3.0            2131.0                    794000.0   
1                 3.0            2860.0                   1529000.0   
2                 3.0            1515.0                    794000.0   
3                 3.0            2508.0                   1422500.0   
4                 2.0            1804.0                   1070000.0   
5                 2.0            1610.0                   1529000.0   
6                 2.0            1035.0                   1070000.0   
7                 3.0            2522.0                    794000.0   
8                 3.0            2623.0                   1070000.0   
9                 1.0             947.0                   1385000.0   

  Median Price Growth (%) in the Area Median Home Price  \
0                                -3.2          753310.0   
1                                -5.6         1641350.0   
2                                -3.2          753310.0   
3                                -6.7         1411000.0   
4                                 0.0         1071000.0   
5                                -5.6         1641350.0   
6                                 0.0         1071000.0   
7                                -3.2          753310.0   
8                                 0.0         1071000.0   
9                                -1.1         1434800.0   

  Monthly Payment (($)[Tax&Ins]) Minimum Qualifying Income ($)  \
0                         4320.0                      129600.0   
1                         9420.0                      282600.0   
2                         4320.0                      129600.0   
3                         8100.0                      243000.0   
4                         6150.0                      184500.0   
5                         9420.0                      282600.0   
6                         6150.0                      184500.0   
7                         4320.0                      129600.0   
8                         6150.0                      184500.0   
9                         8230.0                      246900.0   

  Unemployment Rate (%) City Population Total Crimes per 1000  \
0                  3.41         65483.0                 25.09   
1                  2.20        106020.0                 28.29   
2                  3.41         38692.0                 19.44   
3                  2.39        886007.0                 66.99   
4                  3.15        122788.0                 62.26   
5                  2.20        106020.0                 28.29   
6                  3.15         78907.0                 37.50   
7                  3.41         38692.0                 19.44   
8                  3.15         78907.0                 37.50   
9                  2.35        154859.0                 26.46   

  Violent Crimes per 1000 Property Crimes per 1000 Burglary Crimes per 1000  \
0                    2.54                    20.39                     2.17   
1                    2.51                    20.91                     4.87  

**From the sample of 10 predictions here are the results:**

    Lowest price per sqf - 418 USD, in for 566 sqf lot area, 5 bedrooms/3 bathrooms single family house in city of Martinez, Contra Costa county, with school rating A-. Total value $1,025,000 <br>
    
    Highest predicted price per sqf - 1419 USD, in 5928 sqf lot area, 3 bedrooms/2 bathrooms, single family house in city of San Mateo, San Mateo county, with school rating A+. Total value: $ 2,398,000 <br>
    
    Mid range predicted price per sqf: <br>
    750 USD, in 0 sqf lot area, 2 bedrooms/1 bathrooms, condo in city of Sunnyvale, Santa Clara county, with school rating A+. Total value: $ 759,850 <br>
    758 USD, in 2996.928 sqf lot area, 4 bedrooms/3 bathrooms, single family house in city of San Francisco, San Francisco county, with school rating A-. Total value: $ 1,850,000 <br>

